aten/src/THCS/generic/THCSTensor.cpp

```diff
@@ -133,111 +143,125 @@ THCSTensor *THCSTensor_(new)(THCState *state)
 /* Pointer-copy init */
 THCSTensor *THCSTensor_(newWithTensor)(THCState *state, THCIndexTensor *indices, THCTensor *values)
```

Comment:
Why doesn't this one do the check as well?

Reply:
The check is only relevant when sizes are given (see the old newWithTensorAndSizes, the check is only reached when sizes != none)

Reply:
Ah I see. It makes sense. Thanks for explaining.